### ISSUE TITLE
top insets wrong value not same as safe area insets

### ISSUE THREAD

**Issue by dhruv-00** [jpudysz/react-native-unistyles#170]
### Description

The insets value of top is showing wrong it is not same as useSafeAreaInsets some times it shows fine and some time when app is opened with killed state notificaion

### Steps to reproduce

create new react native app install unistyles and safe area context match both insets value it will be different check in the repro

### Snack or a link to a repository (optional)

https://github.com/dhruv-00/unistyles-repro

### Unistyles version

2.5.1

### React Native version

0.73.6

### Platforms

Android

### Engine

JSC

### Architecture

Paper (old)

**Comment by dhruv-00** [comment]
https://github.com/jpudysz/react-native-unistyles/assets/86352938/111398c6-db21-44a6-b6f8-9d
<img width="808" alt="Screenshot 2024-03-29 at 4 06 52 PM" src="https://github.com/jpudysz/react-native-unistyles/assets/86352938/9a2e2fc2-09a1-4bfd-bf02-075c2b25b5c9">
<img width="1104" alt="Screenshot 2024-03-29 at 4 07 14 PM" src="https://github.com/jpudysz/react-native-unistyles/assets/86352938/917ce58a-f956-498f-aa12-135410621832">

6f561ad6dd

**Comment by jpudysz** [comment]
Hello 👋

But what is the issue? That unistyles is not showing exact same values as `safe-area-context`?

On your example status bar is translucent or not? If not then indeed it's a bug, if it is then Unistyles reports correct value = 46. 

What's your phone model / android version?
What's your setup, do you use expo? Do you use any windows flags?
If you want to me help I need a lot more details.

Also please take a look here: https://www.unistyl.es/reference/dimensions/#insets

**Comment by dhruv-00** [comment]
the issue is that normally it is working fine but when the app is in killed state and I open it with notification then the top value is not proper.
In my main app the issue is coming when opening with notification 
when I tried to create it with new reproducible example then it was already showing wrong value in the first time without notification opening

phone: OnPlus 12 / Android 14
react native cli

**Comment by dhruv-00** [comment]
after setting status bar translucent 
![WhatsApp Image 2024-03-29 at 5 10 15 PM](https://github.com/jpudysz/react-native-unistyles/assets/86352938/b5ffa8ee-684e-406f-9ea8-5e9bdf4bb3a9)

**Comment by jpudysz** [comment]
Ok, so you don't have satus bar translucent, and it happens only if you open the app from push notification?

**Comment by dhruv-00** [comment]
yes exactly, and my main app is react-native 0.72.6 
but in the reproducible example it is latest version but it is reproducible without opening the notification

**Comment by jpudysz** [comment]
Ok to sum it up:

Android 14 with RN 0.73, Unistyles 2.5.1, when you open fresh app you're receiving invalid top inset (but not always).
You're not using expo.

So the last question:
Did you modify anyhow window flags? Example is in the documentation posted above.

**Comment by dhruv-00** [comment]
No I did not change any windows flags

**Comment by jpudysz** [comment]
Cool, thanks for the report. I'm a little bit surprised as Android 14 has a stable inset API. Maybe it's something about OnePlus

**Comment by jpudysz** [comment]
I will be able to get OnePlus with Android 14 on Tuesday (after Easter). I will debug it directly 🙏

**Comment by efstathiosntonas** [comment]
@jpudysz 2.5.2 fixes this or it's unrelated?

**Comment by jpudysz** [comment]
It was not addressed specifically for this issue. I will debug this report tomorrow with OnePlus device 🙏

**Comment by jpudysz** [comment]
Hey @dhruv-00 can you check how it works with fresh release (2.5.2 )?

**Comment by dhruv-00** [comment]
let me check I'll update you

**Comment by jpudysz** [comment]
Ok thanks for the tests. Tomorrow, I will setup deeplinking and try to repro it with OnePlus 🙏

**Comment by jpudysz** [comment]
Tested on OnePlus 9 Pro with Android 14.

1) Opening app from killed state with deeplinking:

```
npx uri-scheme open unistyles:// --android
```

Shows always `top=0`. 

2) I was able to reproduce double reporting with:

```tsx
 <StatusBar translucent />
```

```shell
 (NOBRIDGE) LOG  Bridgeless mode is enabled
 (NOBRIDGE) LOG  Running "bare" with {"rootTag":51,"initialProps":{},"fabric":true}
 (NOBRIDGE) LOG  UnistylesRuntime.insets: 0
 (NOBRIDGE) LOG  UnistylesRuntime.insets: 32
```

But it's because `StatusBar` is rendered in JSX, so it happends after first re-render.

Any other idea how can I test it?

**Comment by dhruv-00** [comment]
On which react native version you are trying it?

**Comment by jpudysz** [comment]
0.74-rc.5 and 0.73.6 in your example

**Comment by jpudysz** [comment]
Tried your example with 0.72.6:

![image](https://github.com/jpudysz/react-native-unistyles/assets/9088288/e2a18fd8-5904-4895-b4da-96a009a2790a)

hmm, any ideas whats different in your app?

**Comment by dhruv-00** [comment]
first one is with notification and second one is with uri scheme
https://github.com/jpudysz/react-native-unistyles/assets/86352938/942aee07-4b4c-4a01-8280-933a5d2cb8c5

**Comment by jpudysz** [comment]
Ok, so I need to setup Firebase project 🫡

**Comment by dhruv-00** [comment]
Are there any steps by which I can debug this? 
may be looking into any specific files.
which you think might be causing the issue.

**Comment by jpudysz** [comment]
You would need to debug kotlin files.. 
Let me firstly try with some push notification. I just hope I will be able to repro and isolate the issue

**Comment by jpudysz** [comment]
Reproduced!

![image](https://github.com/jpudysz/react-native-unistyles/assets/9088288/54efe601-9093-44cd-9e37-12019536e411)

**Comment by jpudysz** [comment]
That was extremely difficult to spot, but thanks to you, Unistyles can now properly initialize in headless mode.

The issue was that layout listeners were not being properly set when the app was initializing from headless mode, e.g. from a push notification. You should now receive the correct value.

Disclaimer:
Unistyles will emit two insets events when run in headless mode, but this should not be visible to the end user.
It's possible to emit a single event, but I need to refactor how the C++ module works, so most likely, this will be implemented in Unistyles 3.0.

I will release Unistyles 2.5.3 soon 🎉
